File: WebClient SDK/src/default/files/clientFileTemplatePermissionsGetFileContentExtension.ts
import {
  CardGetFileContentExtension,
  ICardGetFileContentExtensionContext
} from 'tessa/cards/extensions';
import { UIContext } from 'tessa/ui';
import { KrToken } from 'tessa/workflow';
import { ICardModel } from 'tessa/ui/cards';
import { CardTemplateHelper } from 'tessa/cards';

export class ClientFileTemplatePermissionsGetFileContentExtension extends CardGetFileContentExtension {
  public beforeRequest(context: ICardGetFileContentExtensionContext): void {
    const editor = UIContext.current.cardEditor;
    let model: ICardModel;
    let token: KrToken;

    if (
      editor &&
      (model = editor.cardModel!) &&
      (token = KrToken.tryGet(model.card.info)!) &&
      context.request.versionRowId === CardTemplateHelper.ReplacePlaceholdersVersionRowID &&
      model.card.id === context.request.info[CardTemplateHelper.PlaceholderCurrentCardIDInfo]
    ) {
      token.setInfo(context.request.info);
    }
  }
}

File: Source/Extensions/Tessa.Extensions.Default.Client/Files/ClientFileTemplatePermissionsGetFileContentExtension.cs
using System;
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Cards.Extensions;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.Platform.Storage;
using Tessa.UI;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.Files
{
    public sealed class ClientFileTemplatePermissionsGetFileContentExtension :
        CardGetFileContentExtension
    {
        #region Base Overrides

        public override Task BeforeRequest(ICardGetFileContentExtensionContext context)
        {
            ICardEditorModel editor = UIContext.Current.CardEditor;
            ICardModel model;
            KrToken token;

            if (editor is not null
                && (model = editor.CardModel) is not null
                && (token = KrToken.TryGet(model.Card.Info)) is not null
                && context.Request.VersionRowID == CardHelper.ReplacePlaceholdersVersionRowID
                && model.Card.ID == context.Request?.Info.TryGet<Guid?>(CardHelper.PlaceholderCurrentCardIDInfo))
            {
                token.Set(context.Request.Info);
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}
